src/modules/doctor/doctor.service.ts

```diff
@@ -110,4 +112,58 @@ export default class DoctorService {
       throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
+
+  async updateDoctorAvailability(
+    doctorId: number,
+    availabilities: Availability[],
+  ): Promise<Availability[]> {
+    try {
+      const doctor = await this.doctorRepository
+        .createQueryBuilder('doctor')
+        .where('doctor.id = :id', { id: doctorId })
+        .getOne();
+
+      if (!doctor) {
+        throw new NotFoundException(`Doctor with id ${doctorId} not found`);
```

Comment:
find other way. typeorm query builder have a specific function to do

Reply:
rewrited with findOneOrFail function